Public/Src/Cache/ContentStore/Library/Service/Grpc/GrpcCopyClient.cs

```diff
@@ -24,28 +26,85 @@ namespace BuildXL.Cache.ContentStore.Service.Grpc
     /// </summary>
     public sealed class GrpcCopyClient : IShutdown<BoolResult>
     {
+        // Link to DistributedContentStoreSettings.MaxConcurrentCopyOperations
+        private const int _maxClientCount = 512;
+
+        private static readonly object lockObject = new object();
+        internal static readonly ConcurrentDictionary<(string, int, bool), GrpcCopyClient> _clientDict = new ConcurrentDictionary<(string, int, bool), GrpcCopyClient>(Environment.ProcessorCount, _maxClientCount);
+        private static Task _backgroundCleaningTask;
+        private static CancellationTokenSource _backgroundCleaningTaskTokenSource;
+
         private readonly Channel _channel;
         private readonly ContentServer.ContentServerClient _client;
         private readonly string _host;
         private readonly int _grpcPort;
         private bool _useCompression;
+        internal DateTime _lastUseTime;
+        internal int _uses;
+
+        static GrpcCopyClient()
+        {
+            RestartBackgroundCleanup();
+        }
+
+        private async static Task BackgroundCleanupAsync(CancellationTokenSource cts)
+        {
+            _backgroundCleaningTaskTokenSource = cts;
+            var ct = _backgroundCleaningTaskTokenSource.Token;
+
+            while (!ct.IsCancellationRequested)
+            {
+                if (!_clientDict.IsEmpty)
+                {
+                    var oneHourAgo = DateTime.UtcNow - TimeSpan.FromHours(1);

```

Comment:
>FromHours [](start = 64, length = 9)

This is border line close to the timeout of 30 minutes. Don't we run into a chance here where some times flip between just before the 1/2 hour timeout and some just after? Wouldn't it be better to have something relatively deterministic in the nr of cycles of the cleanup? i.e. 55 minutes or a 14 minute wait in the cleanup?